lib/internal/cluster/master.js

```diff
@@ -98,12 +104,29 @@ function setupSettingsNT(settings) {
 function createWorkerProcess(id, env) {
   const workerEnv = util._extend({}, process.env);
   const execArgv = cluster.settings.execArgv.slice();
-  const debugArgRegex = /--inspect(?:-brk|-port)?|--debug-port/;
 
   util._extend(workerEnv, env);
   workerEnv.NODE_UNIQUE_ID = '' + id;
 
-  if (execArgv.some((arg) => arg.match(debugArgRegex))) {
+  // This allows user to override inspect port for workers.
+  const clusterDebugArgs = execArgv
+    .filter((arg) => arg.match(debugArgRegex))
+    .sort();
+
+  let debugSettingsOverriden = false;
```

Comment:
Another option, make `cluster.settings.execArgv` a getter/setter property and do check-on-assignment